### PR TITLE
Miner/Sharder update settings tests

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**
Cases added:

Miner settings update:
- Miner update min_stake by delegate wallet should work
- Miner update num_delegates by delegate wallet should work
- Miner update max_stake with delegate wallet should work
- Miner update multiple settings with delegate wallet should work
- Miner update min_stake with less than global min stake should fail
- Miner update num_delegates greater than global max_delegates should fail
- Miner update max_stake greater than global max_stake should fail
- Miner update max_stake less than min_stake should fail
- Miner update min_stake negative value should fail
- Miner update max_stake negative value should fail
- Miner update num_delegate negative value should fail
- Miner update without miner id flag should fail
- Miner update with nothing to update should fail
- Miner update settings from non-delegate wallet should fail

Sharder settings update:
- Sharder update min_stake by delegate wallet should work
- Sharder update num_delegates by delegate wallet should work
- Sharder update max_stake by delegate wallet should work
- Sharder update multiple settings with delegate wallet should work
- Sharder update with min_stake less than global min should fail
- Sharder update with num_delegates more than global max_delegates should fail
- Sharder update max_stake more than global max_stake should fail
- Sharder update min_stake greater than max_stake should fail
- Sharder update min_stake negative value should fail
- Sharder update max_stake value should fail
- Sharder update num_delegates negative value should fail
- Sharder update without sharder id flag should fail
- Sharder update with nothing to update should fail
- Sharder update settings from non-delegate wallet should fail

Bug fixes:
- https://github.com/0chain/0chain/pull/856
- https://github.com/0chain/0chain/pull/861
- https://github.com/0chain/0chain/pull/869

### CODE REVIEW

**Comment by stewartie4** [comment]
Amazing coverage - really extensive 
One last case I can think of 
./zwallet mn-user-info -> i don't think we have coverage for this?

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | miner-update-tests  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | moreMaxStakePatches  |
| blobber  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by Kishan-Dhakan** [comment]
> Amazing coverage - really extensive One last case I can think of ./zwallet mn-user-info -> i don't think we have coverage for this?

Thanks Ryan! `./zwallet mn-user-info` fetches the pools of tokens locked against Miner or Sharder node. Since it would be out of context in this ticket, I can raise a new PR for it.

File: tests/cli_tests/zwalletcli_sharder_update_settings_test.go [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+			"min_stake": 1,
+		}), true)
+		require.Nil(t, err, "error reverting sharder node settings after test")
+		require.Len(t, output, 1)
+		require.Equal(t, "settings updated", output[0])
+
+		output, err = minerInfo(t, configPath, createParams(map[string]interface{}{
+			"id": sharder.ID,
+		}), true)
+		require.Nil(t, err, "error fwtching sharder info")
```
Comment (littlestar642) [inline tests/cli_tests/zwalletcli_sharder_update_settings_test.go:71]: rectify the typo here

**Comment by littlestar642** [comment]
Hey @Kishan-Dhakan 
Really nice coverage. I liked how you refactored the defer out of all the tests. I think this pattern can be added to blobber config update tests as well.

File: tests/cli_tests/zwalletcli_sharder_update_settings_test.go (hunk 1 above)
Comment (Kishan-Dhakan) [reply tests/cli_tests/zwalletcli_sharder_update_settings_test.go]: Fixed

**Comment by Kishan-Dhakan** [comment]
> Hey @Kishan-Dhakan Really nice coverage. I liked how you refactored the defer out of all the tests. I think this pattern can be added to blobber config update tests as well.

Thanks Avinash!

**Pull request merged**